crates/ruff_linter/src/rules/ruff/rules/falsy_dict_get_fallback.rs

```diff
@@ -0,0 +1,179 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{AlwaysFixableViolation, Applicability, Diagnostic, Edit};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{
+    Expr, ExprAttribute, ExprBooleanLiteral, ExprCall, ExprDict, ExprList, ExprName,
+    ExprNoneLiteral, ExprNumberLiteral, ExprSet, ExprStringLiteral, Int, Number,
+};
+use ruff_python_semantic::analyze::typing;
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::{Ranged, TextRange};
+
+/// ## What it does
+/// Checks for `dict.get(key, falsy_value)` used as implicit casts to boolean values.
+///
+/// ## Why is this bad?
+/// The default fallback `None` is already falsy.
+///
+/// ## Example
+///
+/// ```python
+/// if dict.get(key, False):
+///     ...
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// if dict.get(key):
+///     ...
+/// ```
+///
+/// ## Fix safety
+/// This rule's fix is marked as safe, unless the `dict.get()` call contains comments between arguments.
+#[derive(ViolationMetadata)]
+pub(crate) struct FalsyDictGetFallback;
+
+impl AlwaysFixableViolation for FalsyDictGetFallback {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Avoid providing a falsy fallback to `dict.get()` when used in a boolean context. The default fallback `None` is already falsy.".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        "Remove falsy fallback from `dict.get()`".to_string()
+    }
+}
+
+pub(crate) fn falsy_dict_get_fallback(checker: &mut Checker, expr: &Expr) {
+    // Check if the expression is a call
+    let Expr::Call(ExprCall { func, .. }) = expr else {
+        return;
+    };
+
+    // Check if the function being called is an attribute (e.g. `dict.get`)
+    let Expr::Attribute(ExprAttribute { value, attr, .. }) = &**func else {
+        return;
+    };
+
+    // Ensure the method called is `get`
+    if attr != "get" {
+        return;
+    }
+
+    // Check if the object is a dictionary using the semantic model
+    if let Expr::Name(expr_name) = &**value {
+        if !is_known_to_be_of_type_dict(checker.semantic(), expr_name) {
+            return;
+        }
+    } else {
+        return;
+    }
+
+    let Expr::Call(ExprCall { arguments, .. }) = expr else {
+        return;
+    };
+
+    // Check if a fallback arg is provided
+    if arguments.args.len() < 2 {
+        return;
+    }
+
+    // Get the fallback argument
+    let fallback_arg = &arguments.args[1];
+
+    // Define what is considered a falsy value
+    let is_falsy = is_falsy_fallback(fallback_arg);
+
+    if is_falsy {
+        let diagnostic = Diagnostic::new(FalsyDictGetFallback, fallback_arg.range());
+
+        if let Some(arg) = arguments.args.get(1) {
```

Comment:
Can we use the `remove_argument` helper here https://github.com/astral-sh/ruff/blob/b63c2e126b928fab4180eda5d9132552beeee7fe/crates/ruff_linter/src/fix/edits.rs#L207